citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -314,14 +342,31 @@ private PageRequest createPageRequest(IUser user, int page, String sortBy, ICita
         request.setLastUpdated(OffsetDateTime.now());
         return request;
     }
-    
+
     @Override
     public List<BibField> getItemTypeFields(IUser user, ItemType itemType) {
         return zoteroManager.getFields(user, itemType);
     }
-    
+
     @Override
     public List<String> getValidCreatorTypes(IUser user, ItemType itemType) {
         return zoteroManager.getValidCreatorTypes(user, itemType);
     }
+
+    @Override
+    public boolean deleteCitation(IUser user, String groupId, ICitation citation) throws ZoteroConnectionException,
+            GroupDoesNotExistException, ZoteroHttpStatusException, ZoteroItemDeletionFailedException {
+        Optional<CitationGroup> groupOptional = groupRepository.findById(new Long(groupId));
+        if (!groupOptional.isPresent()) {
+            throw new GroupDoesNotExistException();
+        }
+
+        zoteroManager.deleteCitation(user, groupId, citation);
+        citationRepository.delete((Citation) citation);
```

Comment:
this will probably fail if citation is referenced from any page request

Reply:
what should i do about it

Reply:
delete all the references to it first